core/templates/dev/head/pages/library-page/search-bar/search-bar.directive.ts

```diff
@@ -47,6 +47,200 @@ angular.module('oppia').directive('searchBar', [
             HtmlEscaperService, LanguageUtilService, NavigationService,
             SearchService, UrlService, SEARCH_DROPDOWN_CATEGORIES) {
           var ctrl = this;
+          /**
+           * Opens the submenu.
+           * @param {object} evt
+           * @param {String} menuName - name of menu, on which
+           * open/close action to be performed (category,language).
+           */
+          ctrl.openSubmenu = function(evt, menuName) {
+            NavigationService.openSubmenu(evt, menuName);
+          };
+          /**
+           * Handles keydown events on menus.
+           * @param {object} evt
+           * @param {String} menuName - name of menu to perform action
+           * on(category/language)
+           * @param {object} eventsTobeHandled - Map keyboard events('Enter')
+           * to corresponding actions to be performed(open/close).
+           *
+           * @example
+           *  onMenuKeypress($event, 'category', {enter: 'open'})
+           */
+
```

Comment:
Drop this line, because the docstring above should be associated with the function.